python/ray/tune/suggest/ax.py

```diff
@@ -124,8 +124,14 @@ def __init__(self,
         assert ax is not None, """Ax must be installed!
             You can install AxSearch with the command:
             `pip install ax-platform sqlalchemy`."""
+        
+        ax_mode_is_min = ax.experiment.optimization_config.objective.minimize 
+        expected_mode = "min" if ax_mode_is_min else "max"
         if mode:
-            assert mode in ["min", "max"], "`mode` must be 'min' or 'max'."
+            assert mode in ["min", "max"], "`mode` must be 'min' or 'max'."         
+            assert mode == expected_mode, "`mode` does not correspond to the one set in `AxClient`."
```

Comment:
Let's move this check to the end of `_setup_experiment()`. This way we can still do this check if`mode` and `metric` is passed directly to `tune.run()`.